internal/uvm/virtual_device.go

```diff
@@ -6,20 +6,41 @@ import (
 
 	"github.com/Microsoft/go-winio/pkg/guid"
 	"github.com/Microsoft/hcsshim/internal/guestrequest"
+	"github.com/Microsoft/hcsshim/internal/log"
 	"github.com/Microsoft/hcsshim/internal/requesttype"
 	hcsschema "github.com/Microsoft/hcsshim/internal/schema2"
 )
 
-func (uvm *UtilityVM) AssignDevice(ctx context.Context, device hcsschema.VirtualPciDevice) (string, error) {
+// VPCIDevice represents a vpci device. Holds its guid and a handle to the uvm it
+// belongs to.
+type VPCIDevice struct {
+	vm *UtilityVM
+	ID string
+}
+
+// Release frees the resources of the corresponding vpci device
+func (vpci *VPCIDevice) Release(ctx context.Context) error {
+	if vpci.vm == nil {
```

Comment:
I'm curious why we have these `vm == nil` checks on all the `Release` functions. Are there cases where we expect to call `Release` on a resource that doesn't have a VM associated?